src/lib/autoDetect.ts

```diff
@@ -0,0 +1,73 @@
+import { ExtensionContext, workspace, window, languages } from 'vscode';
+import { FileAssociations } from './types';
+const { extname } = require('path');
+
+export const initializeHubLAutoDetect = (context: ExtensionContext) => {
+  languages.setLanguageConfiguration('hubl', {
+    indentationRules: {
+      decreaseIndentPattern: /.*{%(.*?end).*%}*./,
+      increaseIndentPattern: /.*{%(?!.*end).*%}.*/,
+    },
+  });
+  const languageExtensions = {
+    '.html': 'html-hubl',
+    '.css': 'css-hubl',
+  };
+  const getHublPreference = context.workspaceState.get('USE_HUBL');
```

Comment:
Yeah, I think the case of the feedback form -- we don't want that to pop up if developers are entering various projects. Its a one and done deal. However, this _could_ be seen as a setting that should be prompted in each workspace. 

I don't use workspaces either, but I think in this context the "workspace" is the current open project/filesystem in VSCode - since this is working for me when I test it randomly in a theme's folder.

My only question about the workspace vs global state settings is - if we can set the user's settings instead of the workspace for the associations -- does that seem generally like what a developer would want for all their projects? Do we prompt again after they say "yes" to the workspace settings to see if they want to set it on a global level instead?

Reply:
> I don't use workspaces either, but I think in this context the "workspace" is the current open project/filesystem in VSCode - since this is working for me when I test it randomly in a theme's folder.

Yeah. IMO you don't want this in the global settings otherwise every HTML file you open in VSCode would be identified as HubL + HTML, which would definitely get annoying for anyone who works on things besides HubSpot websites. 

I believe writing to the workspaceState (and thus creating/editing the `.vscode` folder) is the correct way to persist this type of state: https://code.visualstudio.com/api/extension-capabilities/common-capabilities#data-storage.

Reply:
Some thoughts on how this should work in slack - https://hubspot.slack.com/archives/C043D9UJRHP/p1690492323365829